src/sidebar/components/select-filter.js

```diff
@@ -0,0 +1,69 @@
+import { createElement } from 'preact';
+import propTypes from 'prop-types';
+
+import Menu from './menu';
+import MenuItem from './menu-item';
+import SvgIcon from '../../shared/components/svg-icon';
+
+/**
+ * @typedef {import('../store/modules/filters').FilterOption} FilterOption
+ */
+
+/**
+ * @typedef SelectFilterProps
+ * @prop {FilterOption} defaultOption
+ * @prop {string} [icon]
+ * @prop {(selectedFilter: FilterOption) => any} onSelect
+ * @prop {FilterOption[]} options
+ * @prop {FilterOption} [selectedOption]
+ * @prop {string} title
+ */
+
+/**
+ * A select-element-like control for selecting one of a defined set of
+ * options.
+ *
+ * @param {SelectFilterProps} props
+ */
+function SelectFilter({
```

Comment:
Is this component intended to be a a foundation generic select that we could use for other purposes?

Can I suggest that the name `FilterSelect` would be more consistent with the `<Entity><Widget Type>` naming scheme for other components (eg. `UserMenu`, `SidebarView`, `TagList`).